### PR TITLE
aws-apigateway-importer 1.0.1 (new formula)

### PR DESCRIPTION
**Pull request by moshen**
New formula for the AWS API Gateway Importer: https://github.com/awslabs/aws-apigateway-importer

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Install is failing, probably needs `ENV["_JAVA_OPTIONS"] = "-Duser.home=#{ENV["HOME"]}"` added at the beginning of `install`.

**Comment by moshen** [comment]
According to http://bot.brew.sh/job/Homebrew%20Pull%20Requests/37828/version=el_capitan/injectedEnvVars/ , `HOME` is `/Users/brew`, which `mvn` was trying to write to, [but couldn't (because sandboxing?)](http://bot.brew.sh/job/Homebrew%20Pull%20Requests/37828/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_aws_apigateway_importer/).

Should it set the `mvn` local repository folder to be within the build folder?  [Other `mvn` builds seem to just call `mvn` without resetting environment.](https://github.com/Homebrew/homebrew/blob/5be873292232558110dd6c8d277c5f7f9226316e/Library/Formula/swagger-codegen.rb)

**Comment by moshen** [comment]
Ok, I see @DomT4 suggesting changing `user.home` to `buildpath` in #46729 .

**Comment by DomT4** [comment]
Yeah, we haven't really nailed down the syntax yet. There may be something more unified depending on what happens with https://github.com/Homebrew/homebrew/pull/46777.

**Comment by mikemcquaid** [comment]
Test is failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/37843/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_aws_apigateway_importer/

**Comment by moshen** [comment]
oops... fixed.

File: Library/Formula/aws-apigateway-importer.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  def install
+    ENV["_JAVA_OPTIONS"] = "-Duser.home=#{buildpath}"
+    system "mvn", "assembly:assembly"
+    libexec.install "target/aws-apigateway-importer-1.0.1-jar-with-dependencies.jar"
+    bin.write_jar_script libexec/"aws-apigateway-importer-1.0.1-jar-with-dependencies.jar", "aws-api-import"
+  end
+
+  test do
+    (`#{bin}/aws-api-import` =~ /Usage:\s+aws-api-import/) == 0
```
Comment (mikemcquaid) [inline Library/Formula/aws-apigateway-importer.rb]: Use `assert_match` and `shell_output` here.

Comment (mikemcquaid) [inline Library/Formula/aws-apigateway-importer.rb]: Also: can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

Comment (moshen) [inline Library/Formula/aws-apigateway-importer.rb]: Unfortunately, all the options either read from or modify stuff on an aws account.  `--help` is also the same as calling with no options, but it couldn't hurt to be more specific.

Comment (moshen) [inline Library/Formula/aws-apigateway-importer.rb]: Also, `mvn assembly:assembly` runs the unit tests associated with the project.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by moshen** [comment]
:beers: Thanks!

**Pull request closed without merging**